dandiapi/api/models/asset.py

```diff
@@ -125,25 +126,57 @@ def copy(cls, asset):
         return Asset(path=asset.path, blob=asset.blob, metadata=asset.metadata)
 
     @classmethod
-    def get_path(cls, path_prefix: str, qs: List[str]) -> Set:
+    def get_path(cls, dandiset_pk: str, version: str, path_prefix: str, qs: List[Asset]) -> Set:
         """
         Return the unique files/directories that directly reside under the specified path.
 
         The specified path must be a folder (must end with a slash).
         """
-        if not path_prefix:
-            path_prefix = '/'
         prefix_parts = [part for part in path_prefix.split('/') if part]
-        paths = set()
+        paths = {'folders': {}, 'items': {}}
+
+        # Declare base api url and kwargs for all asset reverse lookups
+        base_api_url = 'https://api.dandiarchive.org'
+        kwargs = {'versions__dandiset__pk': dandiset_pk, 'versions__version': version}
+
         for asset in qs:
-            path_parts = [part for part in asset['path'].split('/') if part]
+            path_parts = [part for part in asset.path.split('/') if part]
 
             # Pivot index is -1 (include all path parts) if prefix is '/'
             pivot_index = path_parts.index(prefix_parts[-1]) if len(prefix_parts) else -1
             base_path, *remainder = path_parts[pivot_index + 1 :]
-            paths.add(f'{base_path}/' if len(remainder) else base_path)
-
-        return sorted(paths)
+            is_folder = len(remainder)
+
+            if is_folder:
+                if paths['folders'].get(base_path) is None:
+                    paths['folders'][base_path] = {
+                        'path': f'{base_path}/',
+                        'size': 0,
+                        'num_items': 0,
+                        'created': asset.created,
+                        'modified': asset.modified,
+                    }
+
+                entry = paths['folders'][base_path]
+                entry['size'] += asset.size
+                entry['num_items'] += 1
+                entry['created'] = min(entry['created'], asset.modified)  # earlist
+                entry['modified'] = max(entry['modified'], asset.modified)  # latest
+            else:
+                kwargs['asset_id'] = asset.asset_id
+                asset_delete_url = base_api_url + reverse('asset-detail', kwargs=kwargs)
+                asset_download_url = base_api_url + reverse('asset-download', kwargs=kwargs)
+
+                paths['items'][base_path] = {
+                    'path': f'{base_path}/',
```

Comment:
```suggestion
                    'path': base_path,
```
why asset should have a trailing `/`?

Reply:
This was more for display on the client, but I agree it's unnecessary, since the client can already tell that it's a folder from the parent key anyway. Should we even include this `path` field at all, since it's ultimately redundant?

Reply:
In 4b8bd5f I've removed the `path` field. If it's important that we keep that field included, I can revert the change.